File: output/LLaMa_repaired_1/avatar/Java/Python/codeforces_86_A.py
def power(a):
    count = 0
    while a > 0:
        a = a // 10
        count += 1
    return count

def mult(a):
    pow = power(a)
    max_val = 0
    for _ in range(pow):
        max_val = max_val * 10 + 9
    return a * (max_val - a)

def get_values(l, r):
    maxxes = []
    temp = 0
    for _ in range(10):
        temp = temp * 10 + 9
        maxxes.append(temp // 2 * (temp - temp // 2))
    
    res = max(mult(l), mult(r))
    
    for _ in range(10):
        temp = temp * 10 + 9
        if l <= temp // 2 and temp // 2 <= r:
            res = max(maxxes[-1], res)
        if l <= temp and temp <= r:
            res = max(temp // 2 * (temp - temp // 2), res)
        maxxes.append(temp // 2 * (temp - temp // 2))
    
    return res, maxxes

l = list(map(int, input().split()))
r = list(map(int, input().split()))

print(get_values(l[0], l[1])[0])

File: output/LLaMa_repaired_1/avatar/Python/Java/codeforces_86_A.java
import java.util.Scanner;

public class codeforces_86_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int l = Integer.parseInt(input[0]);
        int r = Integer.parseInt(input[1]);
        int digitCount = 0;
        int curr = r;
        while (curr != 0) {
            digitCount++;
            curr /= 10;
        }
        int first = (int) Math.pow(10, digitCount);
        int second = first / 2;
        int ans = -1;
        for (int i : new int[]{l, r, first, second}) {
            if (i >= l && i <= r) {
                StringBuilder reverse = new StringBuilder();
                int num = i;
                while (num != 0) {
                    int digit = num % 10;
                    reverse.append(9 - digit);
                    num /= 10;
                }
                long reversedNum = Long.parseLong(reverse.reverse().toString());
                ans = Math.max(ans, reversedNum * i);
            }
        }
        System.out.println(ans);
    }
}